test/dotnet-add-reference.Tests/GivenDotnetAddReference.cs

```diff
@@ -693,5 +693,55 @@ public void WhenIncompatibleFrameworkDetectedItPrintsError(string frameworkArg)
             cmd.StdErr.Should().MatchRegex(" - net45");
             net45lib.CsProjContent().Should().BeEquivalentTo(csProjContent);
         }
+
+        [Fact]
+        public void WhenDirectoryContainingProjectIsGivenReferenceIsAdded()
+        {
+            var setup = Setup();
+            var lib = NewLibWithFrameworks(dir: setup.TestRoot);
+
+            var result = new AddReferenceCommand()
+                    .WithWorkingDirectory(setup.TestRoot)
+                    .WithProject(lib.CsProjPath)
+                    .Execute($"\"{Path.GetDirectoryName(setup.ValidRefCsprojPath)}\"");
+
+            result.Should().Pass();
+            result.StdOut.Should().Be(string.Format(CommonLocalizableStrings.ReferenceAddedToTheProject, @"ValidRef\ValidRef.csproj"));
+            result.StdErr.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void WhenDirectoryContainsNoProjectsItCancelsWholeOperation()
```

Comment:
Why do you say it cancels the whole operation?

Reply:
Simply following a convention from another fact in this test case.  It'll error before it adds any projects, which is why it considers the whole operation to be "canceled".